packages/javascript/bh-shared-ui/src/components/ExploreTable/ManageColumnsComboBox/ManageColumnsComboBox.tsx

```diff
@@ -43,16 +43,12 @@ export const ManageColumnsComboBox = ({
 
     useOnClickOutside(ref, () => setIsOpen(false));
 
-    const initialColumns = useMemo(
+    const selectedColumns = useMemo(
         () => allColumns.filter((item) => selectedColumnsProp[item.id]),
         [allColumns, selectedColumnsProp]
     );
-    useEffect(() => {
-        setSelectedColumns(initialColumns);
-    }, [initialColumns]);
 
     const pinnedColumns = useMemo(() => allColumns.filter((item) => item.isPinned), [allColumns]);
-    const [selectedColumns, setSelectedColumns] = useState<ManageColumnsComboBoxOption[]>(initialColumns);
```

Comment:
Looks like `selectedColumns` can be determined state from the selectedColumnsProp and we can skip the duplicate state updates and treats this component as always controlled. This investigation spurred from a "uncontrolled component becoming controlled" warning in test logs!

Reply:
Agreed! Im thinking the internal state couldve been added when we had the `useEffect` listening to internal state to reset columns to default if non were selected, but im not totally sure!

Ive tested default behavior, reset to default button, select all/none button, and manually unselecting all to reset to default and all are working as expected 👍